python/lib/dcoscli/dcoscli/marathon/main.py

```diff
@@ -856,6 +867,34 @@ def version_list(self, app_id, max_count):
         emitter.publish(versions)
         return 0
 
+    def delay_reset(self, app_id, pod_id):
+        """
+        :param app_id: the id of the application to reset delay
+        :type app_id: str
+        :param pod_id: the Marathon ID of the pod to reset delay
+        :type pod_id: str
+        :return: process return code
+        :rtype: int
+        """
+        response = self._create_marathon_client().delay_reset(app_id or pod_id)
+        if response:
+            emitter.publish(response)
```

Comment:
this is either a string or a dict... while we do emit JSON... it is only when the user uses the `--json` flag.

Also... it should NOT be a `return 0` below if this is a dict... only if a string.

Reply:
although there are a number of places where this code naively returns 0... I believe that is because an exception could be thrown... in which case python will caught the exception and return a non-zero.   (I would have to dig deeper to confirm).   This would not be the case in this code.   I believe the code flow might be easier if you have `204` return a value and anything else throw an exception.  Those the return of that function is only a `str`.